federationapi/routing/join.go

```diff
@@ -253,11 +253,12 @@ func SendJoin(
 	// there isn't much point in sending another join event into the room.
 	alreadyJoined := false
 	for _, se := range stateAndAuthChainResponse.StateEvents {
+		if !se.StateKeyEquals(*event.StateKey()) {
```

Comment:
I suspect this might panic if `event.StateKey()` returns `nil`?

Reply:
This is checked already above (line 168)

Reply:
So it is!

Reply:
That's not ideal, relying on a check roughly 100 lines above. This function is far too long :/